server/src/main/java/org/opensearch/rest/action/search/RestSearchAction.java

```diff
@@ -180,6 +180,12 @@ public static void parseSearchRequest(
             searchRequest.allowPartialSearchResults(request.paramAsBoolean("allow_partial_search_results", null));
         }
 
+        if (request.hasParam("phase_took")) {
+            // only set if we have the parameter passed to override the cluster-level default
+            // else phaseTookQueryParamEnabled = null
+            searchRequest.setPhaseTookQueryParamEnabled(request.paramAsBoolean("phase_took", true));
```

Comment:
This is correct. We are making the "default" true only if `request.hasParam("phase_took")`. This is to handle the 1st case below. If no param is passed, the default value for `phaseTookQueryParamEnabled` is false declared in SearchRequest.
```
_search?phase_took
_search?phase_took=true
_search?phase_took=false
```

Reply:
Okay, so if user passes just phase_took, we are going to treat it as true, got it. Does other param in search or other API have similar behaviour?

Reply:
There are instances of both behaviors. There are 2 paramAsBoolean() functions. First one automatically treats empty string as true. However we are calling the second paramAsBoolean() which does not. I think it is preferable to use true as default value.

```
    @Override
    public boolean paramAsBoolean(String key, boolean defaultValue) {
        String rawParam = param(key);
        // Treat empty string as true because that allows the presence of the url parameter to mean "turn this on"
        if (rawParam != null && rawParam.length() == 0) {
            return true;
        } else {
            return Booleans.parseBoolean(rawParam, defaultValue);
        }
    }

    @Override
    public Boolean paramAsBoolean(String key, Boolean defaultValue) {
        return Booleans.parseBoolean(param(key), defaultValue);
    }
```